From bd7cedffe45fcdd84c0735a4b9c80c17d86468b1 Mon Sep 17 00:00:00 2001
From: Mike Snitzer <snitzer@redhat.com>
Date: Fri, 13 Jun 2014 14:35:32 -0400
Subject: [block] blk-mq: user (1 << order) to implement order_to_size()

Message-id: <1402670238-13916-45-git-send-email-snitzer@redhat.com>
Patchwork-id: 83803
O-Subject: [RHEL7.1 PATCH 044/150] blk-mq: user (1 << order) to implement order_to_size()
Bugzilla: 1105204
RH-Acked-by: David Milburn <dmilburn@redhat.com>
RH-Acked-by: Jeff Moyer <jmoyer@redhat.com>

BZ: 1105204

Upstream commit 4ca085009f447abc1c818f8802e908e9afdd77fa
Author: Ming Lei <tom.leiming@gmail.com>
Date:   Sat Apr 19 18:00:18 2014 +0800

    blk-mq: user (1 << order) to implement order_to_size()

    Cc: Joerg-Volker Peetz <jvpeetz@web.de>
    Cc: Max Filippov <jcmvbkbc@gmail.com>
    Signed-off-by: Ming Lei <tom.leiming@gmail.com>
    Signed-off-by: Jens Axboe <axboe@fb.com>

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/block/blk-mq.c b/block/blk-mq.c
index c277797..3a560a4 100644
--- a/block/blk-mq.c
+++ b/block/blk-mq.c
@@ -1130,12 +1130,7 @@ static void blk_mq_free_rq_map(struct blk_mq_tag_set *set,
 
 static size_t order_to_size(unsigned int order)
 {
- size_t ret = PAGE_SIZE;
-
- while (order--)
-  ret *= 2;
-
- return ret;
+ return (size_t)PAGE_SIZE << order;
 }
 
 static struct blk_mq_tags *blk_mq_init_rq_map(struct blk_mq_tag_set *set,
-- 
1.7.1